src/main/kotlin/org/jetbrains/webstorm/lang/WebAssemblyReference.kt

```diff
@@ -0,0 +1,44 @@
+package org.jetbrains.webstorm.lang
+
+import com.intellij.lang.ASTNode
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.*
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiTreeUtil
+import org.jetbrains.webstorm.lang.psi.*
+
+class WebAssemblyReference(node: ASTNode,
+                           idxNode: ASTNode,
+                           private val namedElements: Array<WebAssemblyNamedElement>?)
```

Comment:
usually, it is better to encapsulate "resolution" logic inside of the reference in the ``multiResolve `` method instead of passing the resolved elements into the constructor.  Because reference can be used not only for the resolution